crypto/ec/ec_curve.c

```diff
@@ -3207,36 +3207,47 @@ int EC_curve_nist2nid(const char *name)
  * This can be used when a curve is loaded explicitly (without a curve
  * name) or to validate that domain parameters have not been modified.
  *
- * Returns: The nid associated with the found named curve or O if not found.
+ * Returns: The nid associated with the found named curve, or NID_undef
+ *          if not found. If there was an error it returns -1.
  */
 int ec_curve_nid_from_params(const EC_GROUP *group)
 {
-    int ret = 0, nid, len, field_type, group_ord_bytes_len;
+    int ret = -1, nid, len, field_type, group_ord_bytes_len, rv;
     size_t i, seed_len;
     const unsigned char *seed, *params_seed, *params_order;
-    unsigned char group_ord_bytes[MAX_CURVE_ORDER_LEN];
+    unsigned char *group_ord_bytes = NULL;
     const EC_CURVE_DATA *data;
     const EC_METHOD *meth;
     EC_GROUP *tmp = NULL;
     BN_CTX *ctx = NULL;
 
     meth = EC_GROUP_method_of(group);
     if (meth == NULL)
-        return 0;
+        return -1;
     /* Use the optional named curve nid as a search field */
     nid = EC_GROUP_get_curve_name(group);
     field_type = EC_METHOD_get_field_type(meth);
     seed_len = EC_GROUP_get_seed_len(group);
     seed = EC_GROUP_get0_seed(group);
-    group_ord_bytes_len = (EC_GROUP_get_degree(group) + 7) / 8;
+
+    len = BN_num_bytes(group->order);
+    group_ord_bytes_len = BN_num_bytes(group->field);
+    /* some of the p values are zero padded in the internal prime tables */
+    if (len > group_ord_bytes_len)
+        group_ord_bytes_len = len;
```

Comment:
I have changed the variable name and updated the comment.
I am not so keen on calling the same function multiple times instead of storing it into a local. I think the comment is sufficient.